comptests/ha/test_ha_shutdown_cluster.py

```diff
@@ -72,41 +72,40 @@ def setup_class(cls):
         LOGGER.info("STARTED: Setup Operations")
         LOGGER.info("Check the overall status of the cluster.")
-        self.restored = True
         resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
         if not resp[0]:
             resp = self.ha_obj.restart_cluster(self.node_master_list[0])
             assert_utils.assert_true(resp[0], resp[1])
         LOGGER.info("Cluster status is online.")
-        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=common_const.HA_POD_NAME_PREFIX)
-        pod_name = pod_list[0]
-        output = self.node_master_list[0].execute_cmd(
-            cmd=common_cmd.KUBECTL_GET_POD_CONTAINERS.format(pod_list[0]),
-            read_lines=True)
-        container_list = output[0].split()
-        for container in container_list:
-            res = self.node_master_list[0].send_k8s_cmd(
-                operation="exec", pod=pod_name, namespace=common_const.NAMESPACE,
-                command_suffix=f"-c {container} -- {common_cmd.SERVICE_HA_STATUS}", decode=True)
-            assert_utils.assert_true(res, "HA services are not running")
+        LOGGER.info("Checking if all the ha services are up and running")
+        resp = HAK8SCompLib.check_ha_services(self.node_master_list[0])
+        assert_utils.assert_true(resp, "HA services are not running")
         LOGGER.info("Done: Setup operations.")
 
     def teardown_method(self):
         """
         This function will be invoked after each test function in the module.
         """
         LOGGER.info("STARTED: Teardown Operations.")
-        if self.restored:
-            LOGGER.info("Cleanup: Check cluster status and start it if not up.")
-            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
-            if not resp[0]:
-                LOGGER.debug("Cluster status: %s", resp)
-                resp = self.ha_obj.restart_cluster(self.node_master_list[0])
-                assert_utils.assert_true(resp[0], resp[1])
+        if self.restore_pod:
+            resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                           restore_method=self.restore_method,
+                                           restore_params={"deployment_name": self.deployment_name,
+                                                           "deployment_backup":
+                                                               self.deployment_backup})
+            LOGGER.debug("Response: %s", resp)
+            assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+            LOGGER.info("Successfully restored pod by %s way", self.restore_method)
+        LOGGER.info("Cleanup: Check cluster status and start it if not up.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        if not resp[0]:
+            LOGGER.debug("Cluster status: %s", resp)
+            resp = self.ha_obj.restart_cluster(self.node_master_list[0])
```

Comment:
In case of test failure, if you restart cluster, you will loose state of cluster.

Reply:
As this is shutdown cluster we need to start it after test execution. Added a bool variable to perform this only if the TC pass.